schematic/models/validate_attribute.py

```diff
@@ -868,186 +883,590 @@ def url_validation(
+    ) -> tuple([[], []]):
+        """Merge two series to identify gather all invalid values, and parse out invalid rows and entries
+        Args:
+            series_1, pd.core.series.Series: first set of invalid values to extract
+            series_2, pd.core.series.Series: second set of invalid values to extract
+        Returns:
+            invalid_rows, list: invalid rows taken from both series
+            invalid_entry, list: invalid values taken from both series
+        """
+        # Merge series to get invalid values and rows
+        invalid_values = pd.merge(series_1, series_2, how="outer")
+        # Increment pandas row values by 2 so they will match manifest sheet row values.
+        invalid_rows = (
+            pd.merge(
+                series_1,
+                series_2,
+                how="outer",
+                left_index=True,
+                right_index=True,
+            ).index.to_numpy()
+            + 2
+        )
+        # Parse invalid_rows and invalid_values so they can be used to raise warnings/errors
+        invalid_rows = np_array_to_str_list(invalid_rows)
+        invalid_entry = iterable_to_str_list(invalid_values.squeeze())
+        return invalid_rows, invalid_entry
+
+    def _format_invalid_row_values(
+        invalid_values: Dict[str, pd.core.series.Series]
+    ) -> tuple([[], []]):
+        """Parse invalid_values dictionary, to extract invalid_rows and invalid_entry to be used later
+        to raise warnings or errors.
+        Args:
+            invalid_values, Dict[str, pd.core.series.Series]:
+        Returns:
+            invalid_rows, list: invalid rows recorded in invalid_values
+            invalid_entry, list: invalid values recorded in invalid_values
+        """
+        # Increment pandas row values by 2 so they will match manifest sheet row values.
+        invalid_rows = invalid_values.index.to_numpy() + 2
+
+        # Parse invalid_rows and invalid_values so they can be used to raise warnings/errors
+        invalid_rows = np_array_to_str_list(invalid_rows)
+        invalid_entry = iterable_to_str_list(invalid_values)
+        return invalid_rows, invalid_entry
+
+    def _gather_set_warnings_errors(
         val_rule: str,
-        manifest_col: pd.core.series.Series,
-        project_scope: list,
+        source_attribute: str,
+        set_validation_store: list[{}, [], {}],
         dmge: DataModelGraphExplorer,
-        access_token: str,
-    ) -> list[list[str]]:
+    ) -> tuple([[], []]):
+        """Based on the cross manifest validation rule, and in set rule scope, pass variables to
+        _get_cross_errors_warnings
+            to log appropriate error or warning.
+        Args:
+            val_rule, str: Validation Rule
+            source_attribute, str: Source manifest column name
+            set_validation_store, list[{},[],{}]: contains the missing_manifest_log, present_manifest_log,
```

Comment:
I'm curious why these three things are saved as one variable. The name isn't very helpful.

Reply:
Okay, so im storing sets of validation outputs that correspond to the scope (since this is how they are bundled to collect the data and to raise errors later). I did this bc otherwise I would have to either return 3 different variables (in order to still bundle by scope, but not package in a list), or output all 6 then import them where relevant. I did this just to simplify the amount of imports/exports. Its called `set_validation_store` bc its the set scope validation store, there is a `value_validation_store` and they are both output as `validation_store`, but for the import I changed the name so it would be more clear the type of validation store that is expected...
Does that make sense?